NuRadioMC/SignalProp/Simple_radiopropa_tracer.py

```diff
@@ -0,0 +1,956 @@
+        self.__shower_axis = shower_axis
+
+    def set_cut_viewing_angle(self,cut):
+        """
+        Set a cut on the viewing angle around the cherenkov angle. Rays with a viewing 
+        angle out of this range will be to dim and won't be seen --> limiting computing time
+
+        Parameters
+        ----------
+        cut: float, default unit
+             range around the cherenkov angle
+        """
+        self.__cut_viewing_angle = cut
+
+    def set_maximum_trajectory_length(self,max_traj_length):
+        """
+        Set a cut on the trajectory length. Otherwise computing infinite may be possible
+
+        Parameters
+        ----------
+        max_traj_length: float, default units
+                         maxmimal length to trace a ray. tracing aborted when reached
+        """
+        self.__max_traj_length = max_traj_length
+
+    def RadioPropa_raytracer(self,n_reflections=0):
+        """
+        Uses RadioPropa to find all the numerical ray tracing solutions between sphere x1 and x2.
+        If reflections is bigger than 0, also bottom reflected rays are searched for with a max
+        of n_reflections of the bottom
+        """
+        try:
+            x1 = self.__x1  * (radiopropa.meter/units.meter)
+            x2 = self.__x2  * (radiopropa.meter/units.meter)
+        except TypeError: 
+            self.__logger.error('NoneType: start or endpoint not initialized')
+            raise TypeError('NoneType: start or endpoint not initialized')
+
+      
+        v = (self.__x2-self.__x1)
+        u = copy.deepcopy(v)
+        u[2] = 0
+        theta_direct, phi = hp.cartesian_to_spherical(*v) # zenith and azimuth for the direct linear ray solution (radians)
+        cherenkov_angle = np.arccos(1. / self.__medium.get_index_of_refraction(self.__x1))
+        
+        ## regions of theta with posible solutions (radians)
+        launch_lower = [0]
+        launch_upper = [theta_direct + np.deg2rad(5)] ## below theta_direct no solutions are possible without upward reflections
+
+        if n_reflections > 0:
+            if self.medium.reflection is None:
+                self.__logger.error("a solution for {:d} reflection(s) off the bottom reflective layer is requested, but ice model does not specify a reflective layer".format(n_reflections))
+                raise AttributeError("a solution for {:d} reflection(s) off the bottom reflective layer is requested, but ice model does not specify a reflective layer".format(n_reflections))
+            else:
+                z_refl = self.__medium.reflection
+                rho_channel = np.linalg.norm(u)
+                rho_bottom = (rho_channel*(z_refl-self.__x2[2]))/(2*z_refl-self.__x2[2]-self.__x1[2])
+                alpha = np.arctan((self.__x1[2]-z_refl)/rho_bottom)
+                ## when reflection on the bottom are allowed, a initial region for theta from 180-alpha to 180 degrees is added
+                launch_lower.append((np.pi - (alpha - np.deg2rad(5))))
```

Comment:
We just needed an offset and this was implemented very early on. Maybe there is a smarter way in doing it but I'm not sure because it is kind of arbitrary.

Reply:
@sjoerd-bouma you are absolutly right, thanks! I'll fix the sign

Reply:
Luckily I keep those calculations :)
![IMG_20210407_133502](https://user-images.githubusercontent.com/65966730/113860888-d8a52900-97a6-11eb-9e84-72b7099284ab.jpg)

Reply:
![image](https://user-images.githubusercontent.com/78698543/113866141-3e94af00-97ad-11eb-8a08-e9d7c51f44f8.png)
So I think you should be scanning not from \pi-\alpha -> \pi, but from \pi - (\pi/2 - \alpha) -> \pi... Do you agree?

I think for the case you've drawn, with the channel higher up than the vertex (and an ice profile that's symmetric in \rho), this is indeed a 'safe' boundary; theoretically, I suppose that if your vertex is higher than the channel, one could have a smaller angle than this, but this would be rare, and I think the safety margin of 5 degrees (when implemented correctly ;) should cover this.

Reply:
Oh god, you're absolutely 100% right. XD

And indeed I only calculated this for a vertex below the channel. Is only a minor adaptation to check whether this is the case and include he other case as well.